server/api/controllers/question.js

```diff
@@ -0,0 +1,24 @@
+const { Question } = require("../models/question");
+
+exports.get_questions = (req, res) => {
+    const limit = Number(req.query.limit);
+
+    if (!limit) {
+        res.status(404).json({
+            error: 'limit is not provided'
+        });
+    }
+
+    Question.count()
+        .exec()
+        .then(count => {
+            const random = Math.floor(Math.random() * count);
+            return Question.find()
+                .skip(random)
+                .limit(limit)
```

Comment:
What do you want to accomplish by using skip and limit?

Reply:
Interesting. But after skipping a certain number of questions we would not have a mix of random questions. Let's say for random numbers 1, 2, 3, 4, 5, 6 and following the questions sequence will be same (For 1: 2, 3, 4, 5, 6 ::::  2: 3, 4, 5, 6, 7 ) see how questions overlap.

Why don't we try getting the `limit` number of unique random numbers within `count` and then return those set of questions?